### PR TITLE
Disambiguate frame id in latencylogger

### PR DESCRIPTION
**Pull request by lukaspetersson**
If the same service for different frames overlap, there is ambiguity in when assigning frame id to a timestamp. In latencylogger, warn the user about this. The user can provide the frame id from the timestamp in these situations.

### CODE REVIEW

File: selfdrive/camerad/cameras/camera_qcom2.cc [hunk 1]
```diff
@@ -1084,10 +1084,10 @@ void process_road_camera(MultiCameraState *s, CameraState *c, int cnt) {
   if ((c == &s->road_cam && env_send_road) || (c == &s->wide_road_cam && env_send_wide_road)) {
     framed.setImage(get_frame_image(b));
   }
-  LOGT("%s: Image set", c == &s->road_cam ? "RoadCamera" : "WideRoadCamera");
+  LOGTF(c->buf.cur_frame_data.frame_id, "%s: Image set", c == &s->road_cam ? "RoadCamera" : "WideRoadCamera");
   if (c == &s->road_cam) {
     framed.setTransform(b->yuv_transform.v);
-    LOGT("%s: Transformed", c == &s->road_cam ? "RoadCamera" : "WideRoadCamera");
+    LOGTF(c->buf.cur_frame_data.frame_id, "%s: Transformed", c == &s->road_cam ? "RoadCamera" : "WideRoadCamera");
```
Comment (gijskoning) [inline selfdrive/camerad/cameras/camera_qcom2.cc:1090]: Log happens if 'c == &s->road_cam'
```suggestion
    LOGTF(c->buf.cur_frame_data.frame_id, "%s: Transformed", "RoadCamera");
```

Comment (gijskoning) [inline selfdrive/camerad/cameras/camera_qcom2.cc:1090]: Log only happens if 'c == &s->road_cam'
```suggestion
    LOGTF(c->buf.cur_frame_data.frame_id, "%s: Transformed", "RoadCamera");
```

File: selfdrive/common/swaglog.h [hunk 2]
```diff
@@ -12,16 +12,21 @@
 void cloudlog_e(int levelnum, const char* filename, int lineno, const char* func,
                 const char* fmt, ...) /*__attribute__ ((format (printf, 6, 7)))*/;
 
-void cloudlog_t(int levelnum, const char* filename, int lineno, const char* func,
+void cloudlog_te(int levelnum, const char* filename, int lineno, const char* func,
                 const char* fmt, ...) /*__attribute__ ((format (printf, 6, 7)))*/;
```
Comment (adeebshihadeh) [inline selfdrive/common/swaglog.h:16]: wrong indentation

File: selfdrive/common/swaglog.h [hunk 3]
```diff
@@ -12,16 +12,21 @@
...
 
+void cloudlog_te(int levelnum, const char* filename, int lineno, const char* func,
+                uint32_t frame_id, const char* fmt, ...) /*__attribute__ ((format (printf, 6, 7)))*/;
```
Comment (adeebshihadeh) [inline selfdrive/common/swaglog.h:19]: wrong indentation

File: selfdrive/common/swaglog.cc [hunk 4]
```diff
@@ -112,15 +100,43 @@ void cloudlog_e(int levelnum, const char* filename, int lineno, const char* func
...
+  int ret = vasprintf(&msg_buf, fmt, args);
   va_end(args);
+  if (ret <= 0 || !msg_buf) return;
+  cloudlog_common(levelnum, filename, lineno, func, msg_buf);
 }
 
-void cloudlog_t(int levelnum, const char* filename, int lineno, const char* func,
-                const char* fmt, ...){
+void cloudlog_t_common(int levelnum, const char* filename, int lineno, const char* func,
+                uint32_t frame_id, const char* fmt, va_list args){
```
Comment (adeebshihadeh) [inline selfdrive/common/swaglog.cc:111]: wrong indentation

File: selfdrive/common/swaglog.cc [hunk 5]
```diff
@@ -112,15 +100,43 @@ void cloudlog_e(int levelnum, const char* filename, int lineno, const char* func
...
   if (!LOG_TIMESTAMPS) return;
+  char* msg_buf = nullptr;
+  int ret = vasprintf(&msg_buf, fmt, args);
+  if (ret <= 0 || !msg_buf) return;
+  json11::Json::object tspt_j = json11::Json::object{ 
+    {"event", msg_buf},
+    {"time", std::to_string(nanos_since_boot())}
+  };
+  if (frame_id > -1) {
```
Comment (adeebshihadeh) [inline selfdrive/common/swaglog.cc:120]: how does this work? frame_id is a uint32_t

File: selfdrive/common/swaglog.cc [hunk 6]
```diff
@@ -112,15 +100,43 @@ void cloudlog_e(int levelnum, const char* filename, int lineno, const char* func
...
+    tspt_j["frame_id"] = std::to_string(frame_id);
+  };
```
Comment (adeebshihadeh) [inline selfdrive/common/swaglog.cc:122]: no semicolon here

File: selfdrive/common/swaglog.cc [hunk 7]
```diff
@@ -112,15 +100,43 @@ void cloudlog_e(int levelnum, const char* filename, int lineno, const char* func
...
+  tspt_j = json11::Json::object{{"timestamp", tspt_j}};
+  cloudlog_common(levelnum, filename, lineno, func, msg_buf, tspt_j);
+}
+
+
+void cloudlog_te(int levelnum, const char* filename, int lineno, const char* func,
+                const char* fmt, ...){
+  va_list args;
+  va_start(args, fmt);
+  cloudlog_t_common(levelnum, filename, lineno, func, -1, fmt, args);
```
Comment (adeebshihadeh) [inline selfdrive/common/swaglog.cc:132]: -1? it's a uint32_t

File: selfdrive/common/swaglog.cc [hunk 8]
```diff
@@ -112,15 +100,43 @@ void cloudlog_e(int levelnum, const char* filename, int lineno, const char* func
...
+  va_end(args);
+}
+void cloudlog_te(int levelnum, const char* filename, int lineno, const char* func,
+                uint32_t frame_id, const char* fmt, ...){
```
Comment (adeebshihadeh) [inline selfdrive/common/swaglog.cc:136]: wrong indentation

File: selfdrive/common/swaglog.cc (hunk 4 above)
Comment (adeebshihadeh) [reply selfdrive/common/swaglog.cc:111]: also add a space between the closing parenthesis and brace

**Pull request merged**